numba_dpex/tests/kernel_tests/test_arg_types.py

```diff
@@ -55,12 +84,28 @@ def check_bool_kernel(A, test):
 
 @pytest.mark.parametrize("filter_str", filter_strings)
 def test_bool_type(filter_str):
-    kernel = dpex.kernel(check_bool_kernel)
+    usm_type = "device"
     a = np.array([2], np.int64)
 
     device = dpctl.SyclDevice(filter_str)
-    with dpctl.device_context(device):
-        kernel[a.size, dpex.DEFAULT_LOCAL_SIZE](a, True)
-        assert a[0] == 111
-        kernel[a.size, dpex.DEFAULT_LOCAL_SIZE](a, False)
-        assert a[0] == 222
+    queue = dpctl.SyclQueue(device)
+
+    da = dpt.usm_ndarray(
+        a.shape,
+        dtype=a.dtype,
+        buffer=usm_type,
+        buffer_ctor_kwargs={"queue": queue},
+    )
+    da.usm_data.copy_from_host(a.reshape((-1)).view("|u1"))
+
+    kernel = dpex.kernel(check_bool_kernel)
+
+    kernel[dpex.Range(a.size)](da, True)
```

Comment:
This test can also be done using a scalar and setting range to `1`, in effect launching a sycl single_task.

Reply:
Also, you can directly use dpctl constructors and do not need to copy data from NumPy.